src/MongoDB/Manager.c

```diff
@@ -281,6 +281,34 @@ static PHP_METHOD(Manager, __construct)
 	}
 } /* }}} */
 
+/* {{{ proto void MongoDB\Driver\Manager::addSubscriber(MongoDB\Driver\Monitoring\Subscriber $subscriber)
+   Registers a monitoring subscriber for this Manager */
+static PHP_METHOD(Manager, addSubscriber)
+{
+	php_phongo_manager_t* intern;
+	zval*                 subscriber;
+
+	PHONGO_PARSE_PARAMETERS_START(1, 1)
+	Z_PARAM_OBJECT_OF_CLASS(subscriber, php_phongo_subscriber_ce)
+	PHONGO_PARSE_PARAMETERS_END();
+
+	intern = Z_MANAGER_OBJ_P(getThis());
+
+	/* Lazily initialize the subscriber HashTable */
```

Comment:
I considered moving the implementation of the add/removeSubscriber methods into functions within `phongo_apm.c`. If so, these would just be responsible for parsing arguments and then delegate to another function. Let me know what you think.

In the case of the Manager class, we'd still need to leave the freeing of `intern->subscribers` in `php_phongo_manager_free_object` so we can't get rid of _all_ of the APM-related code.

Reply:
This is now done.